lib/connection/connection.js

```diff
@@ -427,24 +429,38 @@ function Connection(context) {
    * @returns {String} the query status.
    */
   this.getQueryStatusThrowIfError = async function (queryId) {
-    const status = this.getQueryStatus(queryId);
+    const status = await this.getQueryStatus(queryId);
 
     let message, code, sqlState = null;
 
     if (this.isAnError(status)) {
       const response = await getQueryResponse(queryId);
       message = response['message'] || '';
       code = response['code'] || -1;
-
-      if (response['data']) {
-        message += response['data']['queries'].length > 0 ? response['data']['queries'][0]['errorMessage'] : '';
-        sqlState = response['data']['sqlState'];
+      sqlState = response['state'] || '';
+      let responseData = response['data'];
+
+      if (responseData) {
+        if (responseData['queries'].length > 0) {
```

Comment:
I am not quite sure if we need to change this part. It would be better to wait for the review from @sfc-gh-pmotacki.